src/qibolab/platforms/platform.py

```diff
@@ -30,14 +30,37 @@ def disconnect(self):
         self.design.disconnect()
         self.is_connected = False
 
-    def execute_pulse_sequence(self, sequence, nshots=1024, relaxation_time=None):
+    def execute_pulse_sequence(
+        self, sequence, nshots=1024, relaxation_time=None, fast_reset=False, sim_time=10e-6, acquisition_type=None
```

Comment:
If we actually need all these parameters, it might be worth to think about collecting them into a dedicated structure (e.g. `ExecutionParameters`, or something like that), and carefully document.

While splitting the structure is optional, the documentation is not. So, at least provide an accurate description of all parameters.

Reply:
This will work for me, @stavros11 will that be ok for your drive as well, like will you need more parameters than those ?

Reply:
Something like this ?

```suggestion
        @dataclass
class ExecutionParameters:
    """Data structure to deal with execution parameters
    
    :nshots: Number of shots per point on the experiment
    :relaxation_time: Relaxation time for the qubit
    :fast_reset: Enable or disable fast reset
    :sim_time: Time for the simulation execution 
    :acquisition_type: Data acquisition mode (INTEGRATION, RAW, DISCRIMINATION)
    :averaging_type: Data averaging mode (CYLIC, SINGLESHOT)
    """
    
    nshots: Optional[np.uint32] = None
    relaxation_time: Optional[np.uint32] = None
    fast_reset: Optional[bool] = False
    sim_time: Optional[np.float64] = 10e-6
    acquisition_type: Optional[str] = None
    averaging_type: Optional[str] = None

    def __init__(self, dict):
        for k, v in dict.items():
            setattr(self, k, v)
```